oracle/config/aggregate_market.go

```diff
@@ -0,0 +1,186 @@
+package config
+
+import (
+	"fmt"
+
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+// AggregateMarketConfig represents the market configurations for how currency pairs will
+// be resolved to a final price. Each currency pair can have a list of convertable markets
+// that will be used to convert the price of the currency pair to a common currency pair.
+//
+// Price aggregation broadly follows the following steps:
+//  1. Fetch prices for each currency pair from the price feeds.
+//  2. Calculate the provider-weighted median price for each currency pair.
+//  3. Convert the price of each currency pair to a common currency pair using the
+//     AggregatedFeeds field. If there are multiple convertable feeds for a given currency
+//     pair, calculate a median price for each final currency pair, weighted by the
+//     weights of the convertable feeds.
+type AggregateMarketConfig struct {
+	// Feeds is a map of all of the price feeds that the oracle will fetch prices for along
+	// with the resolution configurations for each feed.
+	Feeds map[string]FeedConfig `mapstructure:"currency_pairs" toml:"currency_pairs"`
+
+	// AggregatedFeeds is the list of valid markets (i.e. price feeds) that can be
+	// used to convert the price of the currency pair to a common currency pair. For
+	// example, if the oracle receives a price for BTC/USDT and USDT/USD, it can use
+	// the conversion market to convert the BTC/USDT price to BTC/USD. These must be
+	// provided in a topologically sorted order that resolve to the same currency pair
+	// defined in the CurrencyPair field.
+	AggregatedFeeds map[string][][]Conversion `mapstructure:"aggregated_feeds" toml:"aggregated_feeds"`
+}
+
+// FeedConfig represents the configurations for a given price feed. Each currency pair
+// will have its own feed configuration.
+type FeedConfig struct {
+	// CurrencyPair is the currency pair that the oracle will fetch prices for.
```

Comment:
At a later time we can add an `Minimum` weight field here that signals how many of the provider's must have returned a price - relative to the weight defined above.